dask_bigquery/core.py

```diff
@@ -174,3 +201,121 @@ def make_create_read_session_request(row_filter="", max_stream_count=0):
+      Credentials for accessing Google APIs. Use this parameter to override
+      default credentials.
+    delete_bucket:
+      Delete bucket in GCS after loading intermediary data to Big Query. The bucket will only be deleted if it
+      didn't exist before.
+      Default: False.
+    parquet_kwargs: dict
+      Additional kwargs to pass to dataframe.write_parquet, such as schema, partition_on or
+      write_index. For writing parquet, pyarrow is required.
+      Default: {"write_index": False, "engine": "pyarrow", "write_metadata_file": False}
+    load_job_kwargs: dict
+      Additional kwargs to pass when creating bigquery.LoadJobConfig, such as schema,
+      time_partitioning, clustering_fields, etc.
+      Default: {"autodetect": True}
+
+    Returns
+    -------
+        LoadJobResult
+    """
+    if project_id is None and hasattr(credentials, "project_id"):
+        # service account credentials have a project associated with them
+        project_id = credentials.project_id
+
+    load_job_kwargs_used = (
+        load_job_kwargs.copy()
+        if load_job_kwargs
+        else {"write_disposition": "WRITE_EMPTY"}
+    )
+    if "schema" not in load_job_kwargs_used:
+        load_job_kwargs_used["autodetect"] = True
+
+    # override the following kwargs, even if user specified them
+    load_job_kwargs_used["source_format"] = bigquery.SourceFormat.PARQUET
+
+    fs = gcs_fs(project_id, credentials=credentials)
+    if fs.exists(bucket):
+        if delete_bucket:
+            # if we didn't create it, we shouldn't delete it
+            warnings.warn(
+                "`delete_bucket=True` can only be used with a non-existing bucket.",
+                category=UserWarning,
+                stacklevel=2,
+            )
+            delete_bucket = False
+    else:
+        fs.mkdir(bucket)
+
+    token = tokenize(df)
+    object_prefix = f"{dt.datetime.utcnow().isoformat()}_{token}"
+    path = f"gs://{bucket}/{object_prefix}"
+
+    parquet_kwargs_used = {"write_index": False}
+    if parquet_kwargs:
+        parquet_kwargs_used.update(parquet_kwargs)
+
+    # override the following kwargs, even if user specified them
+    parquet_kwargs_used["engine"] = "pyarrow"
+    parquet_kwargs_used["write_metadata_file"] = False
+    parquet_kwargs_used.pop("path", None)
+
+    try:
+        df.to_parquet(path, **parquet_kwargs_used)
+
+        with bigquery_client(project_id, credentials=credentials) as client:
+            target_dataset = client.create_dataset(dataset_id, exists_ok=True)
+            target_table = target_dataset.table(table_id)
+
+            job_config = bigquery.LoadJobConfig(
+                **load_job_kwargs_used,
+            )
+            job = client.load_table_from_uri(
+                source_uris=f"{path}/*.parquet",
+                destination=target_table,
+                job_config=job_config,
+            )
+
+        return job.result()
+    except Exception:
```

Comment:
You can remove the except here, if you don't want to do any exception processing steps